packages/material-ui-system/src/styleFunctionSx.js

```diff
@@ -49,22 +25,24 @@ function styleFunctionSx(props) {
     return styles;
   }
 
-  let css = {};
+  const emptyBreakpoints = createEmptyBreakpointObject(theme.breakpoints);
+  const breakpointsKeys = Object.keys(emptyBreakpoints);
+
+  let css = emptyBreakpoints;
 
   Object.keys(styles).forEach((styleKey) => {
     const value = callIfFn(styles[styleKey], theme);
 
     if (typeof value === 'object') {
-      if (filterProps.indexOf(styleKey) !== -1) {
+      if (propToStyleFunction[styleKey]) {
```

Comment:
Check if this is a system prop in contact time, instead of checking with `indexOf`